site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2265,7 +2265,7 @@ public function getBadGradedComponentsCountByGradingSections($g_id, $sections, $
      * @param  string $null_section to check if we look in the null section or not
      * @return int the number of bad submissions
      */
-    private function getBadSubmissionsCount($g_id, $section_key, $is_team, string $null_section) {
+    private function getBadSubmissionsCount(int $g_id, string $section_key, boolean $is_team, string $null_section) {
```

Comment:
You should also add a return type annotation in addition to the parameter types.